applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java

```diff
@@ -869,34 +999,53 @@ private static void assertConfigurationEquals(ConfigurationSetting expected, Res
      *
      * @param expected ConfigurationSetting expected to be returned by the service
      * @param actual ConfigurationSetting contained in the RestResponse body
+     * @return True if the ConfigurationSettings are value equivalent
      */
     private static void assertConfigurationEquals(ConfigurationSetting expected, ConfigurationSetting actual) {
-        if (expected == null) {
-            assertNull(actual);
-            return;
-        }
+        assertTrue(configurationsEqual(expected, actual));
+    }
 
-        assertNotNull(actual);
-        assertEquals(expected.key(), actual.key());
+    /**
+     * Helper method to verify that two ConfigurationSettings are datawise equal.
+     * @param expected ConfigurationSetting expected to be returned by the service.
+     * @param actual ConfigurationSetting returned by the service.
+     * @return True if the ConfigurationSettings are the same setting or their member variables are equivalent.
+     */
+    private static boolean configurationsEqual(ConfigurationSetting expected, ConfigurationSetting actual) {
```

Comment:
iirc, were you planning on writing that equality check for ConfigurationSetting? @maririos wrote one for the .NET version. Would make assertions in our tests cleaner
https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs#L92

Reply:
I would love to add a ConfigurationSetting equals override but there are issues with the getter only properties that will cause a lot of unit tests to fail by using it. If we are okay with adding package private setters for unit testing purposes I will add it.

Reply:
Hmm.. 🤔 I'd be OK with package-private setters. I think it would help consumers be able to do a .equals rather than check properties themselves.

Reply:
We should override the `ConfigurationSetting#equals(Object)` method. It seems like a common-sense thing to do, unless I'm misunderstanding something.